models/product_product.py

```diff
@@ -0,0 +1,7 @@
+from odoo import fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    flower_id = fields.Many2one('flower')
```

Comment:
```suggestion
```

It is completely fine to define the `is_flower` and `flower_id` fields on the template only as it is inherited by `product.product` using delegation inheritance